packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -812,6 +814,34 @@ export class SingleThreadedWallet extends EventEmitter<EventEmitterType>
     this.chainService.registerChannel(channelId, assetHolderAddresses, this);
   }
 
+  async challenge(challengeState: State): Promise<SingleChannelOutput> {
+    const channelId = calculateChannelId(challengeState);
+    const response = WalletResponse.initialize();
+
+    await this.knex.transaction(async tx => {
+      const channel = await this.store.getChannel(channelId, tx);
+      if (!channel) {
+        throw new Error(`No channel found for channel id ${channelId}`);
+      }
+
+      const {objectiveId} = await this.store.ensureObjective(
+        {
+          type: 'SubmitChallenge',
+          participants: [],
+          data: {targetChannelId: channelId, challengeState},
+        },
+        tx
+      );
+
+      await this.store.approveObjective(objectiveId, tx);
+
+      response.queueChannel(channel);
```

Comment:
> What important information is being queued up in the response, here? It looks like we yanked the channel from the DB _only_ to check it exists, and then we queued it up in the response without modifying it all. Is that meaningful?

The reason I'm returning the channel is that the channel result now includes a `challengeStatus`. However, this won't function as expected for `v0` so I think it makes sense to just make it a void method.

>It seems like it might be good to immediately flag that a challenge will be submitted?. Either just internally in the wallet, possibly also in the response, here. That might be important so we can reject updateChannel calls that arrive before the challenge tx is submitted.

As we touched on in standup this will be important in the future but we don't need it for `v0`. Definitely, something we should think about in the future.

Reply:
> The reason I'm returning the channel is that the channel result now includes a challengeStatus. However, this won't function as expected for v0 so I think it makes sense to just make it a void method.

I actually think we want to leave this as is for now. We pass in a `response` to `takeActions` so I think we should return that `response` from the `challenge` call. That means we need to return either a `SingleChannelResult` or `MultipleChannelResult`.